### PR TITLE
[2779] Update Compose Overview documentation

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Update Compose Overview docs with new code snippets and screenshots where necessary.

### 🧪 Testing

Please run Docusaurus to make sure everything renders correctly.

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] ~~New code is covered by unit tests~~
- [ ] ~~Comparison screenshots added for visual changes~~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

spongedoc

### CODE REVIEW

File: docusaurus/docs/Android/04-compose/01-overview.mdx [hunk 1]
```diff
@@ -32,7 +32,7 @@ See the individual pages for these components to learn more about them.
 * [Message List](04-message-components/03-message-list.mdx)
 * [Message Composer](04-message-components/04-message-composer.mdx)
 * [Attachments Picker](04-message-components/05-attachments-picker.mdx)
-* [Selected Message Overlay](04-message-components/06-selected-message-overlay.mdx)
+* [Selected Message Menu](04-message-components/06-selected-message-menu.mdx)
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/01-overview.mdx:36]: There should be also [Channel List Item](03-channel-components/04-default-channel-item.mdx) in the channel components above. I've going to add this line in the PR where I will update the docs for `ChannelListItem`. I will also rename the page as it is now named `DefaultChannelListItem`.

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx [hunk 2]
```diff
@@ -5,7 +5,7 @@ The `MessagesScreen` component is the second screen component in the SDK, which
 * [`MessageListHeader`](./02-message-list-header.mdx): Displays a back button, the name of the channel or thread and a channel avatar.
 * [`MessageList`](./03-message-list.mdx): Loads messages data and shows different states based on the data such as loading, empty and loaded states. Also sets up various action handlers, such as long and single item taps, handles pagination, and provides an option to scroll to the bottom.
 * [`MessageComposer`](./04-message-composer.mdx): Sets up and handles the message input, as well as attachments and message actions like editing and replying.
-* [`SelectedMessageOverlay`](./06-selected-message-overlay.mdx): Shown when the user long taps and selects a message. Allows the user to react to messages and perform different actions, like deleting, editing, replying, starting a thread and more.
+* [`SelectedMessageMenu`](./06-selected-message-menu.mdx): Shown when the user long taps and selects a message. Allows the user to react to messages and perform different actions, like deleting, editing, replying, starting a thread and more.
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:8]: Do you think we need to add also docs for `SelectedReactionsMenu`?

**Review by bychkovdmitry — COMMENTED** [review]
Just few comments not related to this PR

File: docusaurus/docs/Android/04-compose/01-overview.mdx (hunk 1 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/01-overview.mdx:36]: Yeah I agree!

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx (hunk 2 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:8]: We definitely should, since it'll probably be commonly used.

**Review by filbabic — APPROVED** [review]
Can we please update the screenshots with more sensible data and then merge this, the rest makes sense to me and looks good! :]

**Pull request merged**